plugin/evm/vm.go

```diff
@@ -445,19 +445,21 @@ func (vm *VM) repairCanonicalChain() error {
 		return nil
 	}
 
-	go func() {
-		start := time.Now()
-		log.Info("starting to repair canonical chain", "startTime", start)
-		err := vm.chain.WriteCanonicalFromCurrentBlock()
-		if err != nil {
-			log.Error("failed to repair canonical chain", "error", err, "timeElapsed", time.Since(start))
-			return
-		}
-		log.Info("finished repairing canonical chain", "timeElapsed", time.Since(start))
-		if err := vm.db.Put(repairedKey, []byte("finished")); err != nil {
-			log.Error("failed to mark flag for canonical chain repaired", "error", err)
-		}
-	}()
+	start := time.Now()
+	log.Info("starting to repair canonical chain", "startTime", start)
+
+	if err := vm.chain.WriteCanonicalFromCurrentBlock(); err != nil {
+		return fmt.Errorf("failed to repair canonical chain after %v due to: %w", time.Since(start), err)
+	}
+	log.Info("finished repairing canonical chain", "timeElapsed", time.Since(start))
+	if err := vm.db.Put(repairedKey, []byte("finished")); err != nil {
+		return fmt.Errorf("failed to mark flag for canonical chain repaired due to: %w", err)
+	}
+	// TODO at this point, we should be in a valid state, so it may be alright to make this
+	// async
+	if err := vm.chain.ValidateCanonicalChain(); err != nil {
```

Comment:
Once we `WriteCanonicalFromCurrentBlock`, I think it is safe to assume we are in a good place?

Reply:
Yup, gonna convert this PR to a draft until I finalize this. This was just to check that was the case.